mergify_engine/web/api/queues.py

```diff
@@ -319,3 +359,165 @@ async def repository_queues(
             queues.queues.append(queue)
 
         return queues
+
+
+@router.put(
+    "/repos/{owner}/{repository}/queue/{queue_name}/freeze",  # noqa: FS003
+    summary="Freezes merge queue",
+    description="Freezes the merge of the requested queue and the queues following it",
+    response_model=QueueFreezeResponse,
+)
+async def create_queue_freeze(
+    queue_freeze_payload: QueueFreezePayload,
+    application: application_mod.Application = fastapi.Depends(  # noqa: B008
+        security.get_application
+    ),
+    queue_name: rules.QueueName = fastapi.Path(  # noqa: B008
+        ..., description="The name of the queue"
+    ),
+    installation_json: github_types.GitHubInstallation = fastapi.Depends(  # noqa: B008
+        security.get_installation
+    ),
+    repository_ctxt: context.Repository = fastapi.Depends(  # noqa: B008
+        security.get_repository_context_with_queue_freeze_feat_check
+    ),
+) -> QueueFreezeResponse:
+
+    if queue_freeze_payload.reason == "":
+        queue_freeze_payload.reason = "No freeze reason was specified."
+
+    async with github.aget_client(installation_json) as client:
+        repository_ctxt.installation.client = client
```

Comment:
Why do you override the client? You should not need to set up a new client here.

Reply:
Indeed, i did this to solve the client closing, but yielding it solves it way better